docs/api-gateway-v2/03-04-migration.md

```diff
@@ -0,0 +1,154 @@
+      - prefix: /pref/
+      - exact: /no/auth/needed/resource.html
+```
+
+Here is the ApiRule configuration which enforces the same authentication policies:
+
+```yaml
+rules:
+  - path: /no/auth/needed/resource.html
+    methods: ["GET", "POST", "PUT", "DELETE"]
+    accessStrategies:
+    - handler: allow
+  - path: /pref/.*
+    methods: ["GET", "POST", "PUT", "DELETE"]
+    accessStrategies:
+    - handler: jwt
+      config:
+        trusted_issuers:
+        - "https://dex.kyma.local"
+        jwks_urls:
+        - "http://dex-service.kyma-system.svc.cluster.local:5556/keys"
+  - path: /exact/path/to/resource.jpg
+    methods: ["GET", "POST", "PUT", "DELETE"]
+    accessStrategies:
+    - handler: jwt
+      config:
+        trusted_issuers:
+        - "https://auth.kyma.local"
+        jwks_urls:
+        - "http://auth-service.kyma-system.svc.cluster.local:5556/keys"
+  - path: (?!/pref/.*)(?!/exact/path/to/resource.jpg)/.*
+    methods: ["GET", "POST", "PUT", "DELETE"]
+    accessStrategies:
+    - handler: jwt
+      config:
+        trusted_issuers:
+        - "https://dex.kyma.local"
+        - "https://auth.kyma.local"
+        jwks_urls:
+        - "http://dex-service.kyma-system.svc.cluster.local:5556/keys"
+        - "http://auth-service.kyma-system.svc.cluster.local:5556/keys"
+```
+
+Note that the last path configuration contains a regular expression with a negative lookahead. It is used to exclude paths handled by other path settings from the `/.*` path, as there can't be two different configurations for a single path. There is one exception to that, which applies only if the same `excludedPaths` element is present on all `authentication` settings, so the specific path doesn't require any authentication at all. In that case, a rule with `handler: allow` should be created, and the path doesn't have to be excluded using a negative lookahead.
+
+Below is the list showing how ApiRule path value corresponds to the excludedPaths values from Api resource, and what negative lookahead value should be added to the `/.*` path:
+
+| excludedPaths expression type | description | excludedPaths sample value | path value | negative lookahead value |
+|---|---|---|---|---|
+|**exact**| no changes to be made | /exact/path/to/resource.jpg | /exact/path/to/resource.jpg | (?!/exact/path/to/resource.jpg) |
+|**prefix**| add `.*` to the end | /pref/ | /pref/.* | (?!/pref/.*) |
+|**suffix**| add `.*` to the beginning; in negative lookahead add `\s` to the end | /suffix.ico | .*/suffix.ico | (?!.*/suffix.ico\s) |
+|**regex**| no changes to be made | /anything.* | /anything.* | (?!/anything.*) |
+
+When the configuration is ready, please create the APIRule object and test if the service is working as expected on the new host. It should work the same on both hosts.
+
+3. Remove Api resource and dependant resources
+
+```shell script
+kubectl delete api <API_NAME> -n <NAMESPACE>
+kubectl delete virtualservice <API_NAME> -n <NAMESPACE>
```

Comment:
why do we need to manually delete vs? At this point, we don't have api-controller already?

Reply:
Yes. The manual migration should occur after the upgrade and an api-controller is removed during the upgrade.